src/dpp/discordvoiceclient.cpp

```diff
@@ -397,11 +397,34 @@ bool discord_voice_client::is_playing() {
 void discord_voice_client::thread_run()
 {
 	utility::set_thread_name(std::string("vc/") + std::to_string(server_id));
+
+	size_t times_looped;
+	time_t time_since_loop;
+
 	do {
 		bool error = false;
 		ssl_client::read_loop();
 		ssl_client::close();
+
+		time_t current_time = time(nullptr);
+		/* Here, we check if it's been longer than 3 seconds since the previous loop,
+		 * this gives us time to see if it's an actual disconnect, or an error.
+		 * This will prevent us from looping too much, meaning error codes do not cause an infinite loop.
+		 */
+		if(current_time - time_since_loop >= 3) {
```

Comment:
`time_since_loop` is uninitialized here the first time this loop is executed, this is UB

Reply:
Whoops, old habits 🙃